src/sources/mithril.rs

```diff
@@ -0,0 +1,249 @@
+use gasket::framework::*;
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use itertools::Itertools;
+use miette::{Context as _, IntoDiagnostic as _};
+use mithril_client::{ClientBuilder, MessageBuilder, MithrilError, MithrilResult};
+use pallas::{
+    ledger::traverse::MultiEraBlock,
+    network::miniprotocols::Point::{self, *},
+};
+use serde::Deserialize;
+use std::{path::Path, sync::Arc};
+use tracing::warn;
+
+use crate::framework::*;
+
+struct Feedback {
+    _multi: MultiProgress,
+    download_pb: ProgressBar,
+    validate_pb: ProgressBar,
+}
```

Comment:
I don't think we should introduce progressbar type of feedback. Oura's main use-case is to run as a daemon where logging is more important.